packages/mssql/src/index.ts

```diff
@@ -0,0 +1,403 @@
+import * as Pool from "@effect/io/Pool"
+import type { Scope } from "@effect/io/Scope"
+import * as Client from "@sqlfx/sql/Client"
+import type { Connection } from "@sqlfx/sql/Connection"
+import { SqlError } from "@sqlfx/sql/Error"
+import type { Custom, Primitive as _Primitive } from "@sqlfx/sql/Statement"
+import * as Statement from "@sqlfx/sql/Statement"
+import * as transform from "@sqlfx/sql/Transform"
+import * as Tedious from "tedious"
+
+export {
+  /**
+   * Column renaming helpers.
+   *
+   * @since 1.0.0
+   */
+  transform,
+}
+
+/**
+ * @category model
+ * @since 1.0.0
+ */
+export interface TediousClient extends Client.Client {
+  readonly config: TediusClientConfig
+}
+
+/**
+ * @category tag
+ * @since 1.0.0
+ */
+export const tag = Tag<TediousClient>()
+
+/**
+ * @category constructor
+ * @since 1.0.0
+ */
+export interface TediusClientConfig {
+
+  readonly domain?: string
+  readonly server?: string
+  readonly port?: number
+  readonly database?: string
+  readonly username?: string
+  readonly password?: ConfigSecret.ConfigSecret
+  readonly connectTimeout?: Duration
+
+  readonly minConnections?: number
+  readonly maxConnections?: number
+  readonly connectionTTL?: Duration
+
+  readonly transformResultNames?: (str: string) => string
+  readonly transformQueryNames?: (str: string) => string
+}
+
+const escape = Statement.defaultEscape('')
+
+/**
+ * @category constructor
+ * @since 1.0.0
+ */
+export const make = (
+  options: TediusClientConfig,
+): Effect.Effect<Scope, never, TediousClient> =>
+  Effect.gen(function*(_) {
+    const compiler = makeCompiler(options.transformQueryNames)
+
+    const transformRows = Client.defaultRowTransform(
+      options.transformResultNames!,
+    )
+
+    const makeConnection = pipe(
+      Effect.acquireRelease(
+        Effect.tap(Effect.sync(() =>
+          new Tedious.Connection({
+            options: {
+              port: options.port,
+              database: options.database,
+              connectTimeout: options.connectTimeout?.millis
+            },
+            server: options.server,
+            domain: options.domain,
+            authentication: {
+              options: {
+                userName: options.username,
+                password: options.password
+                  ? ConfigSecret.value(options.password)
+                  : undefined,
+              }
+            },
+          }),
+        ), _ => Effect.sync(() => _.connect())),
+        _ =>
+          Effect.async<never, never, void>(resume => {
+            _.on("end", () => resume(Effect.unit()))
+            _.close()
+          },
+          ),
+      ),
+      Effect.map((conn): Connection => {
+        const run = (
+          sql: string,
+          _values?: ReadonlyArray<any>,
+          transform = true,
+          rowsAsArray = false,
+        ) =>
+          Effect.async<never, SqlError, any>(resume => {
+            const req = new Tedious.Request(sql, (error, _rowCount, result) => {
```

Comment:
Very awkward api, the request gets the callback for execution, but isn't the one that executes the query